minio/minio
# Clean validation leaks in notification targets

## Description

This PR fixes two things

- Stop spawning store replay routines when testing the notification targets
- Properly honor the target.Close() to clean the resources used

The problem being - we call this `NewTarget()` function in `TestNotificationTargets()` and in `fetchLambdaInfo`. Each and every NewTarget call will spawn goroutines for event replays. So at one situation there will be more than one set of goroutines doing the replay. We get this error in such cases :-

![image](https://user-images.githubusercontent.com/5410427/72008168-3e8bc380-3279-11ea-9a46-de6497289295.png)

(Note : We will delete the event from the store upon a successful PUT/..)

So, at this point the target.Send() will try to delete an event which should've been already deleted by any concurrent goroutines spun during TestNotificationTarget() calls - (We call this function during `config set`)


## Motivation and Context
Fixes #8707

## How to test this PR?
1 - Configure a bucket to receive event notifications ( take webhook incase )
2 - Start generating many events with an active target - eg - ` for i in {1..9999};do ~/mc cp /etc/issue myminio/testb/yyyy$i.jpg; done`
3 - meanwhile call `mc admin config set` repeatedly - like for eg `for i in {1..15}; mc admin config set myminio/ notify_webhook:1 endpoint="http://localhost:8080/" queue_dir="/tmp/minioevents" queue_limit=9998; done`
4 - You should not see `no such file or directory` error in the console.

## Types of changes

## Checklist:
